.github/workflows/coverage-badge.yml

```
@@ -0,0 +1,43 @@
+        badges-directory: .github/badges
+        generate-branches-badge: true
+        generate-summary: true
+
+    - name: Commit and push badge if changed
+      run: |
+        git config --local user.email "action@github.com"
+        git config --local user.name "GitHub Action"
+        git add .github/badges/
+        git commit -m "Update coverage badge" || echo "No changes to commit"
+        git push
+        git push
```

YAMLlint (1.35.1)

[error] no new line character at the end of file

(new-line-at-end-of-file)